homeassistant/components/synology_dsm/__init__.py

```diff
@@ -221,6 +233,46 @@ async def _async_update_listener(hass: HomeAssistantType, entry: ConfigEntry):
     await hass.config_entries.async_reload(entry.entry_id)
 
 
+async def _async_setup_services(hass: HomeAssistantType):
+    """Service handler setup."""
+
+    async def service_handler(call: ServiceCall):
+        """Handle service call."""
+        _LOGGER.debug(
+            "service_handler - called as '%s' with data: %s", call.service, call.data
+        )
+        serial = call.data.get(CONF_SERIAL)
+        dsm_devices = hass.data[DOMAIN]
```

Comment:
I think here : `hass.data[DOMAIN][entry.unique_id]`, we should add a comment to say "Unique ID should be serial for integration services" (also in config_flow ?)

Reply:
sorry, but not sure if i get it 😕 could you explain a little bit more or provide an example?

Reply:
When we set `entry.unique_id` (config flow: `await self.async_set_unique_id(serial, raise_on_progress=False)`) and save it to the data (init `hass.data[DOMAIN][entry.unique_id] = {`), I think we should add a comment to precise the unique_id should be the serial (to prevent future changes).


```python
    # unique_id should be serial for services purpose
    hass.data[DOMAIN][entry.unique_id] = {
        SYNO_API: api,
        UNDO_UPDATE_LISTENER: undo_listener,
    }
```

Reply:
ah yeah ... now I get and fully agree ... have added the comment in config_flow, because there `unique_id` is set